core/shared/src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -936,6 +955,17 @@ object ZPure extends ZPureLowPriorityImplicits with ZPureArities {
   def accessM[R]: AccessMPartiallyApplied[R] =
     new AccessMPartiallyApplied
 
+  /**
+   * Constructs a computation, catching any `Throwable` that is thrown.
+   */
+  def attempt[S, A](a: => A): ZPure[Nothing, S, S, Any, Throwable, A] =
+    suspend {
+      try ZPure.succeed(a)
+      catch {
+        case NonFatal(e) => ZPure.fail(e)
```

Comment:
We would normally only throw if it is an instance of `VirtualMachineError`.

Reply:
Done. But why??

Reply:
I think minor disagreements over what should be treated as fatal and a desire to do as little work as possible when the system is already in an undefined state.